### DISCUSSION TITLE
Some files can not be saved

### DISCUSSION THREAD

**Discussion by zhuanshenlikai** [eclipse-theia/theia#13636] [Q&A]
In the same project, only a few specific files cannot be saved by Ctrl+S, and even when I try to close those files, they cannot be closed.As shown in the figure below, the files marked with a square can be saved, while the files marked with a circle cannot be saved or closed.I have created a Docker image of Theia and it is running in a container environment.
The Theia version: 1.47.1

![1713942646816](https://github.com/eclipse-theia/theia/assets/12570095/79d51b33-cd47-47cf-8230-1445719a3b77)

**Comment by msujew** [comment]
Hey @zhuanshenlikai,

do you get any errors in the (browser) console? Does this also happen when running on your local system?

**Reply by zhuanshenlikai** [reply]
Hey @msujew,Here's the translation:

So, in the recent version of Theia, a frontendConnectionTimeout parameter has been provided in the backend socket keep-alive mechanism. This parameter is used to set the timeout period for the backend socket to wait for a connection after the frontend socket is disconnected. I have set this timeout period to 5 seconds. Here's how I tested it:

1.Opened a file called demo.js.
2.Disconnected the network and made some changes to demo.js. At this point, the file couldn't be saved because the network was disconnected, which is expected.
3.Reconnected the network and noticed that the unsaved changes in demo.js couldn't be saved anymore.

After multiple attempts, it seems that this is because the time taken to reconnect after the network was disconnected was less than 5 seconds. As a result, the backend continued to reuse the old socket, which caused the inability to save the file. I feel that this mechanism is not reasonable. Even if the backend service reuses the old socket, it should still allow saving after the network is restored, shouldn't it?

**Reply by msujew** [reply]
Alright, the issue only popping up after losing connection is pretty important info that was left out. Sounds like a general reconnection issue. Can you create a bug report with appropriate reproduction steps please?

**Reply by zhuanshenlikai** [reply]
Ok,i have created a bug report about this #13702
